sound/soc/sof/topology.c

```diff
@@ -1728,73 +1728,26 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 			    int type)
 {
 	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
-	struct snd_soc_tplg_private *private = &tw->priv;
 	struct snd_soc_dapm_widget *widget = swidget->widget;
-	const struct snd_kcontrol_new *kc;
-	struct soc_bytes_ext *sbe;
-	struct soc_mixer_control *sm;
-	struct soc_enum *se;
+	struct snd_soc_tplg_private *private = &tw->priv;
 	struct snd_sof_control *scontrol = NULL;
-	struct sof_abi_hdr *pdata = NULL;
 	struct sof_ipc_comp_process *process;
-	size_t ipc_size, ipc_data_size = 0;
-	int ret, i, offset = 0;
+	const struct snd_kcontrol_new *kc;
+	struct soc_bytes_ext *sbe;
+	size_t ipc_size = 0;
+	int ctrl_type;
+	int ipc_cmd;
+	int ret;
+	int i;
 
 	if (type == SOF_COMP_NONE) {
 		dev_err(sdev->dev, "error: invalid process comp type %d\n",
 			type);
 		return -EINVAL;
 	}
 
-	/*
-	 * get possible component controls - get size of all pdata,
-	 * then memcpy with headers
-	 */
-	for (i = 0; i < widget->num_kcontrols; i++) {
-
-		kc = &widget->kcontrol_news[i];
-
-		switch (widget->dobj.widget.kcontrol_type) {
-		case SND_SOC_TPLG_TYPE_MIXER:
-			sm = (struct soc_mixer_control *)kc->private_value;
-			scontrol = sm->dobj.private;
-			break;
-		case SND_SOC_TPLG_TYPE_BYTES:
-			sbe = (struct soc_bytes_ext *)kc->private_value;
-			scontrol = sbe->dobj.private;
-			break;
-		case SND_SOC_TPLG_TYPE_ENUM:
-			se = (struct soc_enum *)kc->private_value;
-			scontrol = se->dobj.private;
-			break;
-		default:
-			dev_err(sdev->dev, "error: unknown kcontrol type %d in widget %s\n",
-				widget->dobj.widget.kcontrol_type,
-				widget->name);
-			return -EINVAL;
-		}
-
-		if (!scontrol) {
-			dev_err(sdev->dev, "error: no scontrol for widget %s\n",
-				widget->name);
-			return -EINVAL;
-		}
-
-		/* don't include if no private data */
-		pdata = scontrol->control_data->data;
-		if (!pdata)
-			continue;
-
-		/* make sure data is valid - data can be updated at runtime */
-		if (pdata->magic != SOF_ABI_MAGIC)
-			continue;
-
-		ipc_data_size += pdata->size;
-	}
-
 	ipc_size = sizeof(struct sof_ipc_comp_process) +
-		le32_to_cpu(private->size) +
-		ipc_data_size;
+		le32_to_cpu(private->size);
```

Comment:
that one I just can't figure out? Looks to me like things may work for binary controls but will no longer work for other types of controls.
Confused.

Reply:
@plbossart see above. but I'm not totally sure about this, it is very easy to put back the parsing for other types of data. @ranj063 what do you think?

Reply:
I don't see what's different between the various types of kcontrols. If you can set them in topology then it should be the same handling for all.

Reply:
@plbossart, ok fine by me, i'll add them back